### PR TITLE
[msal-angular] Redirect component and sample updates

### PR DESCRIPTION
**Pull request by jo-arroyo**
This PR:
- Adds `msal.redirect.component` to msal-angular
- Adds `inProgress$` subject to msal-angular, which broadcasts the InteractionStatus, in line with msal-browser and msal-react
- Updates `angular10-sample-app`, `angular11-sample-app` and `angular11-b2c-sample` to import and bootstrap redirect component and uses `inProgress$` subject
- Updates `angular9-v2-sample-app` to use `setLoginDisplay` instead of `checkAccount` in app.component
- Adds documentation on how to use redirects and `inProgress$` in msal-angular
- Updates known-issues doc re. Angular 9 `canLoad` issue
- Adds logging to all msal-angular-v2 samples

PR #2946 was merged into this PR, and:
- Adds getInteractionStatusFromEvent to msal-browser
- Updates msal-react and msal-angular to use getInteractionStatusFromEvent
- Adds unit tests

The PR addresses issue #2868 and #2835.

### CODE REVIEW

**Comment by eluchsinger** [comment]
Is this related to #2868?

**Comment by jasonnutter** [comment]
> Is this related to #2868?

Yes, we think this component and improved samples will help address the problems in that issue.

File: lib/msal-angular/docs/v2-docs/best-practices.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
+# MSAL Angular 2.x Best Practices
+
+Here are some of our optional but recommended practices when using MSAL Angular 2.x.
+
+## Dedicated redirect component
+
+MSAL Angular 2.x provides a dedicated redirect component that can be incorporated into your application. We recommend using this, as `msal.redirect.component.ts` is designed to handle all redirects for your application without your components needing to call `handleRedirectObservable()` manually. 
+
+### Steps for adding the redirect component with hash routing
```
Comment (eluchsinger) [inline lib/msal-angular/docs/v2-docs/best-practices.md:9]: Hash routing is the [old way](https://angular.io/guide/router#locationstrategy-and-browser-url-styles) of routing. Maybe place the default browsing first and move the hash routing section down.

**Review by eluchsinger — COMMENTED** [review]
Hash routing is the [old way](https://angular.io/guide/router#locationstrategy-and-browser-url-styles) of routing. Maybe place the default browsing first and move the hash routing section down.

File: lib/msal-browser/src/utils/BrowserConstants.ts [hunk 2]
```diff
@@ -81,6 +81,27 @@ export enum InteractionType {
...
+/**
+ * Types of interaction currently in progress.
+ * Used in events in wrapper libraries to invoke functions when certain interaction is in progress or all interactions are complete.
+ */
+export enum InteractionStatus {
+    // Initial status before interaction occurs
+    Startup = "startup",
+    // Status set when all login calls occuring
+    Login = "login",
+    // Status set when logout call occuring
```
Comment (tnorling) [inline lib/msal-browser/src/utils/BrowserConstants.ts:93]: Not sure these individual comments are totally necessary but if you change these from single line comments `//` to block comments `/**/` they'll show up in the typedocs as well.

**Review by tnorling — APPROVED** [review]
One minor, non-blocking suggestion. Otherwise looks good!

File: lib/msal-browser/src/utils/BrowserConstants.ts (hunk 2 above)
Comment (jasonnutter) [reply lib/msal-browser/src/utils/BrowserConstants.ts:93]: Definitely would like them to show up in typedocs and intellisense.

Comment (jo-arroyo) [reply lib/msal-browser/src/utils/BrowserConstants.ts]: Makes sense. Changed it, along with a few others in the same file.

**Review by jasonnutter — APPROVED** [review]

**Pull request merged**